File: Coding Problems - Python/Graphs/997 Find the Town Judge.py
class Solution:
    def findJudge(self, n: int, trust: List[List[int]]) -> int:
        Indegree = [0] * (n+1)
        Outdegree = [0] * (n+1)
        for u, v in trust:
            Outdegree[u] += 1
            Indegree[v] += 1

        for i in range(1, n+1):
            if Outdegree[i] == 0 and Indegree[i] == n-1:
                return i
        return -1 

File: February challenges/Java/997 Find the Town Judge.java
class Solution {
    public int findJudge(int n, int[][] trust) {
        int[] Indegree = new int[n+1];
        int[] Outdegree = new int[n+1];

        for(int[] relation: trust){
            int u = relation[0];
            int v = relation[1];
            Outdegree[u] += 1;
            Indegree[v] += 1;
        }
        for(int i = 1; i <= n; i++){
            if(Outdegree[i] == 0 && Indegree[i] == n-1){
                return i;
            }
        }
        return -1;
    }
}
